docs/proposals/metadata-policy.md

```diff
@@ -0,0 +1,163 @@
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/choosing-scheduler.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# MetadataPolicy and its use in choosing the scheduler in a multi-scheduler system
+
+## Introduction
+
+This document describes a new API resource, `MetadataPolicy`, that configures an
+admission controller to take one or more actions based on a pod's metadata.
+Initially the metadata fields that the predicates can examine are labels and annotations,
+and the actions are to add one or more labels and/or annotations, or to reject the pod.
+In the future other actions might be supported, such as applying an initializer. 
+
+The first use of `MetadataPolicy` will be to decide which scheduler should schedule a pod
+in a [multi-scheduler](multiple-schedulers.md) Kubernetes system. In particular, the
+policy will add the scheduler name annotation to a pod based on an annotation that
+is alredy on the pod that indicates the QoS of the pod.
+(That annotation was presumably set by a simpler admission controller that
+uses code, rather than configuration, to map the resource requests and limits of a pod
+to QoS, and attaches the corresponding annotation.)
+
+We anticipate a  number of other uses for `MetadataPolicy`, such as defaulting for
+labels and annotations, prohibiting/requiring particular labels or annotations, or
+choosing a scheduling policy within a scheduler. We do not discuss them in this doc.
+
+
+## API
+
+```go
+// MetadataPolicySpec defines the configuration of the MetadataPolicy API resource.
+// Every rule is applied, in an unspecified order, but if the action for any rule
+// that matches is to reject the pod, then the pod is rejected without being mutated.
+type MetadataPolicySpec struct {
+	Rules []MetadataPolicyRule `json:"rules,omitempty"`
+}
+
+// If the PolicyPredicate is met, then the PolicyAction is applied.
+// Example rules:
+//    reject pod if label with key X is present (i.e. require X)
+//    reject pod if label with key X is not present (i.e. forbid X)
+//    add label X=Y if label with key X is not present (i.e. default X)
+//    add annotation A=B if pod has annotation C=D or E=F
+type MetadataPolicyRule struct {
+	PolicyPredicate PolicyPredicate `json:"policyPredicate"`
+	PolicyAction PolicyAction `json:policyAction"`
+}
+
+// All criteria must be met for the PolicyPredicate to be considered met.
+type PolicyPredicate struct {
+	// Note that Namespace is not listed here because MetadataPolicy is per-Namespace.	
+	LabelSelector *LabelSelector       `json:"labelSelector,omitempty"`
+	AnnotationSelector *LabelSelector  `json:"annotationSelector,omitempty"`
+}
```

Comment:
I'm assuming by "in the future" you mean what we have here now is OK for now. Let me know if I misinterpreted.